src/cascadia/TerminalSettingsModel/ApplicationState.h

```diff
@@ -0,0 +1,65 @@
+/*++
+Copyright (c) Microsoft Corporation
+Licensed under the MIT license.
+
+Module Name:
+- ApplicationState.h
+
+--*/
+#pragma once
+
+#include "ApplicationState.g.h"
+
+#include <inc/cppwinrt_utils.h>
+#include <til/mutex.h>
+#include <til/throttled_func.h>
+
+// This macro generates all getters and setters for ApplicationState.
+// It provides X with the following arguments:
+//   (type, function name, JSON key, ...variadic construction arguments)
+#define MTSM_APPLICATION_STATE_FIELDS(X) \
+    X(std::unordered_set<winrt::guid>, GeneratedProfiles, "generatedProfiles")

```

Comment:
In theory, more things go here later, right? And you need more Traits to convert them?

Reply:
Yeah all fields will just go into this macro (and into the .idl).
But hopefully we won't need more converters. This new one is just a hack, because I wanted to have an unordered set for later, but I don't think we'll need it anywhere else.